src/components/lib/slm/slm.c

```diff
@@ -312,9 +347,17 @@ int
 slm_thd_wakeup(struct slm_thd *t, int redundant)
 {
 	assert(t);
+	if (unlikely(t->cpuid != cos_cpuid())) {
+		struct slm_ipi_percore *ipi_data = slm_ipi_percore_get(t->cpuid);
+		struct slm_ipi_event    event    = { 0 };
+		event.tid = t->tid;
+		int ret = slm_ipi_event_enqueue(&event, t->cpuid);
+		assert(!slm_ipi_event_empty(t->cpuid));
```

Comment:
This can fail if the dequeue happens quickly on the other core. Good assert for debugging, but not good for the PR, I don't think.